minari/data_collector/data_collector.py

```diff
@@ -425,7 +467,25 @@ def save_to_disk(self, path: str, dataset_metadata: Optional[Dict] = None):
         for key, value in dataset_metadata.items():
             self._tmp_f.attrs[key] = value
 
-        self._buffer.append({key: [] for key in STEP_DATA_KEYS})
+        assert "observation_space" not in dataset_metadata.keys()
+        assert "action_space" not in dataset_metadata.keys()
+
+        action_space_str = serialize_space(self.dataset_action_space)
+        observation_space_str = serialize_space(self.dataset_observation_space)
+
+        self._tmp_f.attrs["action_space"] = action_space_str
+        self._tmp_f.attrs["observation_space"] = observation_space_str
+
+        assert "env_observation_space" not in dataset_metadata.keys()
+        assert "env_action_space" not in dataset_metadata.keys()
+
+        env_action_space_str = serialize_space(self.env_action_space)
+        env_observation_space_str = serialize_space(self.env_observation_space)
```

Comment:
[Probably we already discussed it] is it useful to serialize the env spaces? They can be easily restored from the env

Reply:
Thinking about it I agree. We can maybe add an optional flag to save it in a later PR, but I'll remove it from this PR and let's say for now the default behavior is not to save the serialized env spaces, just the dataset ones.

Reply:
done